syntax = "proto3";


enum rebootReason{
    ESP_RST_UNKNOWN = 0;
    ESP_RST_POWERON =1; 
    ESP_RST_EXT = 2;     
    ESP_RST_SW = 3;     
    ESP_RST_PANIC = 4;  
    ESP_RST_INT_WDT = 5;  
    ESP_RST_TASK_WDT = 6;
    ESP_RST_WDT= 7;
    ESP_RST_DEEPSLEEP = 8; 
    ESP_RST_BROWNOUT = 9; 
    ESP_RST_SDIO = 10;   
}

message deviceHealth {

int32 numberOferrorsSinceLastReboot = 1;

int32 numberOfReboots = 2;

int32 deviceRevision = 3;

rebootReason lastRebootReason = 4;

string buildNumber = 5;

string userStringData = 6;

float userNumericData = 7;

}
